### ISSUE TITLE
Input in TD generates a "Invariant Violation: findComponentRoot(...) : Unable to find element."

### ISSUE THREAD

**Issue by remimarenco** [facebook/react#3811]
Hi all!

I am actually working on a React app ([the app!](https://github.com/remimarenco/JsCoverageQC)) and I have a "Invariant Violation: findComponentRoot(...) : Unable to find element." which is pretty weird...

I have a table with a tbody. Inside this I have a tr > td and inside this td I have an input.

When I render it, I have an Invariant Violation error...do you know where it could come from please?

Here is the malicious code :p:

```javascript
<td className="alignMiddle">
    <input type="checkbox" className="exportCheckbox"/>
</td>
```

Any ideas? Or maybe I need to give more details?

Thanks a lot!

**Comment by spicyj** [comment]
Can you try with the master version of React? You can do this by changing your package.json to have

```
"react": "http://react.zpao.com/builds/master/latest/react.tgz"
```

and running npm install again. You may see a more informative warning.

**Comment by remimarenco** [comment]
Hi spicyj!

Thanks for your help! Unfortunately, the peers dependencies do not allow me to do that :(.

```
npm ERR! peerinvalid The package react does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer react-router@0.13.3 wants react@0.13.x
npm ERR! peerinvalid Peer react-hot-loader@1.2.6 wants react@>=0.11.0
```

Do you have an other idea to have better informative warnings?

**Comment by remimarenco** [comment]
Ok so I hacked a bit the peerdependencies of my others components and got the last version of React.
It gives a lot of very useful and detailed informations!

But I am running into a problem which seems to be quite unresolvable...

React asks me this :
```
Warning: validateDOMNesting(...): <div> cannot appear as a child of <tbody> in this context (div > div > div > span > table > tbody). Check the render method of `GeneExonRow`.
```

BUT, if I remove them, React throw me a :
```
Adjacent XJS elements must be wrapped in an enclosing tag
```

What do you think I can do to let React be happy :p?

**Comment by spicyj** [comment]
Sorry, it's a current limitation that you can't return more than one component from a React render method. Something we want to fix (#2127).

You'll have to refactor your components to make this work right now. If it helps, you can have a plain function (not a React component) return an array of elements but that doesn't necessarily help if you need component-local state like you have in GeneExonRow.

**Comment by remimarenco** [comment]
Thanks! I have had already read this kind of limitations by googling but was stuck while trying to wrap around div's (Thought React would read the div as a wrapper and remove it while rendering...).

I have refactored my code and it works like a charm now! 

Be brave for this limitation ;) and thanks for this framework, this is really a pleasure to cut things into module and get out of the javascript/jQuery soup :D!
